From: Felix Krull <f_krull@gmx.de>
Date: Fri, 27 Apr 2018 07:43:50 -0700
Subject: issue8329

# DP: Fix issue #8329: Don't return the same lists from select.select
# DP: when no fds are changed.
---
 Modules/selectmodule.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/Modules/selectmodule.c b/Modules/selectmodule.c
index 11b93a9..3b70fd1 100644
--- a/Modules/selectmodule.c
+++ b/Modules/selectmodule.c
@@ -280,14 +280,6 @@ select_select(PyObject *self, PyObject *args)
 		PyErr_SetFromErrno(SelectError);
 	}
 #endif
-	else if (n == 0) {
-                /* optimization */
-		ifdlist = PyList_New(0);
-		if (ifdlist) {
-			ret = Py_BuildValue("OOO", ifdlist, ifdlist, ifdlist);
-			Py_DECREF(ifdlist);
-		}
-	}
 	else {
 		/* any of these three calls can raise an exception.  it's more
 		   convenient to test for this after all three calls... but
